Added ford annotation for DataFlagEncoder

boldtrn opened the pull request:
As discussed in #236 fords are not routable with the car profile.

The DataFlagEncoder already allows to route through fords. This PR adds annotations to every ford, notifying users that the route contains a ford. The notification is only seen when looking at the instructions, no actual warning is produced.

Right now every Ford is routable. The question is if we actually want that. In #236 there is a discussion about a couple of attributes, including `depth, seasonal, intermittent`, maybe even more :). These checks are not done by the DataFlagEncoder or the GenericWeighting. Do we want to add that?

TODOS:

- [x] Translate the key `way_contains_ford` or decide on a different key.

karussell commented on core/src/test/java/com/graphhopper/routing/util/DataFlagEncoderTest.java:
```diff
@@ -378,4 +385,28 @@ public void testSpatialId() {
         assertEquals(-1, encoder.getMaxspeed(e4, -1, false), .1);
     }
+
+    private Translation getTranslation(final String msg) {
```
Will TranslationMapTest.SINGLETON work here?

boldtrn replied:
Yes, thanks.

boldtrn commented:
![ford_gh](https://cloud.githubusercontent.com/assets/1553525/23494811/12ce7a44-ff69-11e6-83be-2ddeae1cd2c2.png)

I updated the PR with instructions and updated the Test accordingly.

karussell commented on core/src/main/java/com/graphhopper/routing/util/DataFlagEncoder.java:
```diff
@@ -413,7 +412,7 @@ public long handleWayTags(ReaderWay way, long allowed, long relationFlags) {
 
 
-            if(spatialRuleLookupEnabled()){
+            if (spatialRuleLookupEnabled()) {
```
I've seen this in some other PR but forgot to comment: if we make this method public we have to follow java coding rules: `isSpatialRuleLookupEnabled`

karussell commented:
BTW: with DataFlagEncoder.getTransportModeMap we can even enable/disable fords per request
